client/web/src/components/widgets/inputs/SwatchPairInput.vue

```diff
@@ -127,6 +128,21 @@ export default defineComponent({
 	mounted() {
 		this.$watch("primaryColor", this.updatePrimaryColor, { immediate: true });
 		this.$watch("secondaryColor", this.updateSecondaryColor, { immediate: true });
+
+		registerResponseHandler(ResponseType.UpdateWorkingColors, (responseData: Response) => {
+			const colorData = responseData as UpdateWorkingColors;
+			if (colorData) {
+				const { primary } = colorData;
+				const { secondary } = colorData;
+				// G, B, R
+				// console.log(`PRIMARY: ${primary.red}, ${primary.green}, ${primary.blue}`);
+				// console.log(`SEC: ${secondary.blue}`);
+				const primaryButton = this.getRef<HTMLButtonElement>("primaryButton");
+				const secondaryButton = this.getRef<HTMLButtonElement>("secondaryButton");
+				primaryButton.style.setProperty("--swatch-color", `rgba(${primary.red}, ${primary.green}, ${primary.blue}, ${primary.alpha})`);
+				secondaryButton.style.setProperty("--swatch-color", `rgba(${secondary.red}, ${secondary.green}, ${secondary.blue}, ${secondary.alpha})`);
+			}
+		});
```

Comment:
@Keavon should probably take a look at the vue code but on first glance this seems perfectly fine

Reply:
Thats great

Reply:
Done.